examples/contexts/index.tsx

```diff
@@ -0,0 +1,37 @@
+import { gsx } from "gensx";
+
+interface User {
+  name: string;
+}
+
+// Create a context with a default value
+const UserContext = gsx.createContext<User>({
+  name: "",
+});
+
+type GreetingOutput = string;
+
+// Use the context in a component
+const Greeting = gsx.Component<Record<never, never>, GreetingOutput>(
+  "Greeting",
+  () => {
+    const user = gsx.useContext(UserContext);
+    return `Hello, ${user.name}!`;
+  },
+);
+
+async function main() {
+  // Provide a value to the context
+  const result = await gsx.execute(
+    <UserContext.Provider
```

Comment:
Do you think its worth demonstrating nesting providers, or just leave that up to the docs?

Reply:
Agree we should probably show that you can nest providers and all values are available in the tree.

Reply:
added a short section in the docs